pkg/synopsysctl/cmd_migrate.go

```diff
@@ -53,32 +54,55 @@ var migrateCmd = &cobra.Command{
 			cmd.Help()
 			return fmt.Errorf("this command takes 1 argument")
 		}
+		// validate Synopsys Operator image
+		if _, err := util.ValidateImageString(synopsysOperatorImage); err != nil {
+			return err
+		}
+		// validate busy box image
+		if _, err := util.ValidateImageString(busyBoxImage); err != nil {
+			return err
+		}
 		return nil
 	},
 	RunE: func(cmd *cobra.Command, args []string) error {
-		if _, err := util.ValidateImageString(busyBoxImage); err != nil {
+		// Check if a namespace was provided, else determine the namespace from the cluster
+		namespaceToMigrate := ""
+		if len(namespace) > 0 {
+			namespaceToMigrate = namespace
+		} else {
+			isClusterScoped := util.GetClusterScope(apiExtensionClient)
+			if isClusterScoped {
+				namespaces, err := util.GetOperatorNamespace(kubeClient, metav1.NamespaceAll)
+				if err != nil {
+					return err
+				}
+				if len(namespaces) > 1 {
+					return fmt.Errorf("more than 1 Synopsys Operator found in your cluster. please pass the namespace of the Synopsys Operator that you want to migrate")
+				}
+				namespaceToMigrate = namespaces[0]
+			} else {
+				return fmt.Errorf("namespace of Synopsys Operator must be provided in namespace scoped mode")
+			}
+		}
+		// Migrate the CRDs
+		err := migrate(namespaceToMigrate)
+		if err != nil {
 			return err
 		}
 
-		if len(namespace) > 0 {
-			return migrate(namespace)
+		// Update the Operator Image
+		currOperatorSpec, err := soperator.GetOldOperatorSpec(restconfig, kubeClient, namespaceToMigrate) // Get current Synopsys Operator Spec
+		if err != nil {
+			return err
 		}
-
-		// get operator namespace
-		isClusterScoped := util.GetClusterScope(apiExtensionClient)
-		if isClusterScoped {
-			namespaces, err := util.GetOperatorNamespace(kubeClient, metav1.NamespaceAll)
-			if err != nil {
-				return err
-			}
-
-			if len(namespaces) > 1 {
-				return fmt.Errorf("more than 1 Synopsys Operator found in your cluster. please pass the namespace of the Synopsys Operator that you want to migrate")
-			}
-			return migrate(namespaces[0])
+		newOperatorSpec := *currOperatorSpec          // Make copy
```

Comment:
Why make a copy?  There's no need.

Reply:
EnsureSynopsysOperator() takes in `currOperatorSpec` and `newOperatorSpec`. So I made a copy of the current spec before modifying it so that we don't lose the original. Is there a better way to do it?

Reply:
Nope, missed that.